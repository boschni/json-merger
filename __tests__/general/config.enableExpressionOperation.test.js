const jsonMerger = require("../../dist");
const { testConfig } = require("../../__helpers__");

describe("when config.enableExpressionOperation is set", function () {
  test("to undefined it should not process expression operations", function () {
    const object1 = {
      a: {
        aa: "original",
      },
    };

    const object2 = {
      a: {
        aa: {
          $expression: "'should be the value of /a/aa'",
        },
      },
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ enableExpressionOperation: undefined }),
    );

    expect(result).toMatchSnapshot();
  });

  test("to false it should not process expression operations", function () {
    const object1 = {
      a: {
        aa: "original",
      },
    };

    const object2 = {
      a: {
        aa: {
          $expression: "'should be the value of /a/aa'",
        },
      },
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ enableExpressionOperation: false }),
    );

    expect(result).toMatchSnapshot();
  });

  test("to true it should process expression operations", function () {
    const object1 = {
      a: {
        aa: "original",
      },
    };

    const object2 = {
      a: {
        aa: {
          $expression: "'should be the value of /a/aa'",
        },
      },
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ enableExpressionOperation: true }),
    );

    expect(result).toMatchSnapshot();
  });
});
