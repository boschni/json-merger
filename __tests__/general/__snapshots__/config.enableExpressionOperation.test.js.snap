// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when config.enableExpressionOperation is set to false it should not process expression operations 1`] = `
"{
	"a": {
		"aa": {
			"$expression": "'should be the value of /a/aa'"
		}
	}
}"
`;

exports[`when config.enableExpressionOperation is set to true it should process expression operations 1`] = `
"{
	"a": {
		"aa": "should be the value of /a/aa"
	}
}"
`;

exports[`when config.enableExpressionOperation is set to undefined it should not process expression operations 1`] = `
"{
	"a": {
		"aa": {
			"$expression": "'should be the value of /a/aa'"
		}
	}
}"
`;
